non-nix-build/install_nargo.sh

```diff
@@ -14,10 +12,19 @@ fi
 : ${DEPS_DIR:=$PWD/deps}
 cd "$DEPS_DIR"
 
-rm -rf noir
-git clone https://github.com/blocksense-network/noir
-cd noir
-git checkout $WANTED_NARGO_REVISION
-cargo build --release
-cp ./target/release/nargo $BIN_DIR/
+out=$(PWD=$(realpath ../../) ../find_git_hash_from_lockfile.py noir)
+commit=$(echo "${out}" | grep -v "github.com")
+repo=$(echo "${out}" | grep "github.com")
+folder="noir"
 
+mkdir ${folder} || echo "Folder already exists"
+cd "${folder}"
+if [ $(git rev-parse HEAD) != "${commit}" ]; then
+  cd ../
+  rm -rf "${folder}"
```

Comment:
if it's the same repo, we can fetch/checkout the commit and not replacing/cloning the repo again (i remember @zah did something like this in build scripts before)

if it's different, we can add a remote, but i agree sometimes it might be simpler to understand if it's replaced in this case (and if the repo is not that big)